### PR TITLE
localauth init

### PR DESCRIPTION
**Pull request by KonoganB**

### CODE REVIEW

File: genocrowd/app.py [hunk 1]
```diff
@@ -51,6 +57,7 @@ def create_app(config='config/genocrowd.ini', app_name='genocrowd', blueprints=N
     Flask
         Genocrowd Flask application
     """
+    print('yo')
```
Comment (abretaud) [inline genocrowd/app.py]: zis is debug, please remove t

Comment (xgaia) [reply genocrowd/app.py]: For logging, use `app.logger.debug` for debuging

**Comment by abretaud** [comment]
Ah cool, the tests/lint are running on travis now!
But failing: https://travis-ci.com/annotons/genocrowd/jobs/288796694

File: genocrowd/react/src/navbar.jsx [hunk 2]
```diff
@@ -21,21 +21,77 @@ export default class GenocrowdNavbar extends Component {
...
+        let adminLinks
+        if (this.props.config.user.admin) {
+          adminLinks = (
+            <DropdownItem className="bg-dark" tag="Link">
+              <Link className="nav-link" to="/admin"><i className="fas fa-chess-king"></i> Admin</Link>
+            </DropdownItem>
+          )
+        }
+        let integrationLinks
+        if (!this.props.config.disableIntegration || this.props.config.user.admin) {
```
Comment (abretaud) [inline genocrowd/react/src/navbar.jsx]: See above, disableIntegration is from askomics

File: config/genocrowd.ini.template [hunk 3]
```diff
@@ -19,6 +19,9 @@ display_commit_hash = false
 
 database_path = /tmp/genocrowd/database.db
 
+# Disabled upload and integration for non admin (for external endpoint exploration)
```
Comment (abretaud) [inline config/genocrowd.ini.template]: Looks like a copy/ppste from askomics? not needed for genocrowd

File: genocrowd/api/start.py [hunk 4]
```diff
@@ -53,9 +53,14 @@ def start():
...
     """
     try:
-        starter = Start(current_app, session)
-        starter.start()
-
+        #session.clear()
+        if 'user' in session:
+            user = session['user']
+            logged = True
+            print(user)
```
Comment (abretaud) [inline genocrowd/api/start.py]: remove debug print()

File: genocrowd/libgenocrowd/LocalAuth.py [hunk 5]
```diff
@@ -0,0 +1,351 @@
+"""Contain the Database class"""
```
Comment (abretaud) [inline genocrowd/libgenocrowd/LocalAuth.py]: that's not Database

File: genocrowd/libgenocrowd/LocalAuth.py [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+                error = True
+                error_message = 'New passwords are not identical'
+        else:
+            error = True
+            error_message = 'Empty password'
+
+        return {'error': error, 'error_message': error_message, 'user': user}
+
+
+    def get_all_users(self):
```
Comment (abretaud) [inline genocrowd/libgenocrowd/LocalAuth.py]: Not used, remove it (used in askomics to interact with galaxy I guess)

File: genocrowd/libgenocrowd/LocalAuth.py [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+
+        query = '''
+        UPDATE users
+        SET blocked=?
+        WHERE username=?
+        '''
+
+        database.execute_sql_query(query, (new_status, username))
+
+    def set_quota(self, quota, username):
```
Comment (abretaud) [inline genocrowd/libgenocrowd/LocalAuth.py]: Not used, remove it (used in askomics to interact with galaxy I guess)

File: genocrowd/react/src/navbar.jsx [hunk 8]
```diff
@@ -21,21 +21,77 @@ export default class GenocrowdNavbar extends Component {
...
+
+      
     }
 
     return (
       <div>
         <Navbar color="dark" dark expand="md">
           <div className="container">
-            <NavbarBrand href={this.props.config.proxyPath == "/" ? "/" : this.props.config.proxyPath + "/"}>Genocrowd</NavbarBrand>
+            <NavbarBrand href={this.props.config.proxyPath == "/" ? "/" : this.props.config.proxyPath + "/"}> <img src="../static/logo/logoGenocrowd.png"width= {120} height={30} mode='fit'/></NavbarBrand>
```
Comment (abretaud) [inline genocrowd/react/src/navbar.jsx]: spaces problem: `logoGenocrowd.png"width= {120}` ->`logoGenocrowd.png" width={120}`

File: config/genocrowd.test.ini [hunk 9]
```diff
@@ -1,19 +1,25 @@
...
-footer_message = Test
+# If genocrowd is running under a sub path (like http://example.org/genocrowd, subpath is /genocrowd)
+#reverse_proxy_path =
+
+# subtitle =
+footer_message = Welcome to Genocrowd!
 display_commit_hash = false
 
-database_path = /tmp/genocrowd_test/database.db
+database_path = /tmp/genocrowd/database.db
```
Comment (abretaud) [inline config/genocrowd.test.ini]: I think it was only used in askomics, you can just remove it

File: config/genocrowd.ini.template [hunk 10]
```diff
@@ -26,3 +28,5 @@ github = https://github.com/annotons/genocrowd
 # Sentry dsn to report python and js errors in a sentry instance
 # server_dsn = https://00000000000000000000000000000000@exemple.org/1
 # frontend_dsn = https://00000000000000000000000000000000@exemple.org/2
+
```
Comment (abretaud) [inline config/genocrowd.ini.template]: remove trailing blank lines

File: config/genocrowd.test.ini [hunk 11]
```diff
@@ -22,3 +28,5 @@ github = https://github.com/annotons/genocrowd
 # Sentry dsn to report python and js errors in a sentry instance
 # server_dsn = https://00000000000000000000000000000000@exemple.org/1
 # frontend_dsn = https://00000000000000000000000000000000@exemple.org/2
+
```
Comment (abretaud) [inline config/genocrowd.test.ini]: remove trailing blank lines

File: genocrowd/api/admin/admin.py [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+
+    return jsonify({
+        'error': False,
+        'errorMessage': ''
+    })
+
+
+@admin_bp.route('/api/admin/setquota', methods=["POST"])
+@admin_required
+def set_quota():
```
Comment (abretaud) [inline genocrowd/api/admin/admin.py]: We don't need this method

File: tests/conftest.py [hunk 13]
```diff
@@ -35,6 +43,8 @@ class Client(object):
         Description
     db_path : TYPE
```
Comment (abretaud) [inline tests/conftest.py]: Remove here too

File: tests/conftest.py [hunk 14]
```diff
@@ -47,13 +57,10 @@ def __init__(self, config="config/genocrowd.test.ini"):
         """
         # Config
         self.config = config
-        self.db_path = "/tmp/database.db"
+        self.dir_path = tempfile.mkdtemp(prefix="genotest_")
```
Comment (abretaud) [inline tests/conftest.py]: Is this used? If not, just remove it

File: tests/test_api.py [hunk 15]
```diff
@@ -14,3 +15,71 @@ def test_hello(self, client):
...
+        # Jdoe (admin) logged-
+        client.log_user("jdoe")
+
+        expected_config_jdoe = expected_config_nouser
+        expected_config_jdoe["logged"] = True
+        expected_config_jdoe["user"] = {
+            'id': 1,
+            'ldap': False,
+            'username': "jdoe",
+            'email': "jdoe@askomics.org",
```
Comment (abretaud) [inline tests/test_api.py]: "askomics"

File: tests/test_api.py [hunk 16]
```diff
@@ -14,3 +15,71 @@ def test_hello(self, client):
...
+        # jsmith (non admin) logged
+        client.log_user("jsmith")
+
+        expected_config_jsmith = expected_config_nouser
+        expected_config_jsmith["logged"] = True
+        expected_config_jsmith["user"] = {
+            'id': 2,
+            'ldap': False,
+            'username': "jsmith",
+            'email': "jsmith@askomics.org",
```
Comment (abretaud) [inline tests/test_api.py]: "askomics"

File: genocrowd/react/src/routes/error/error.jsx [hunk 17]
```diff
@@ -15,23 +14,23 @@ export default class ErrorDiv extends Component {
       return null
     }
 
-    let messages = {
-      "404": this.utils.objectHaveKeys(this.props, "customMessages", "404") ? this.props.customMessages["404"] : "404 Not Found",
+    // let messages = {
```
Comment (abretaud) [inline genocrowd/react/src/routes/error/error.jsx]: Is there a reason why you don't use it as in https://github.com/askomics/flaskomics/blob/master/askomics/react/src/routes/error/error.jsx#L53?

File: tests/conftest.py (hunk 13 above)
Comment (abretaud) [reply tests/conftest.py]: ping

File: tests/conftest.py (hunk 14 above)
Comment (abretaud) [reply tests/conftest.py]: ping

File: genocrowd/react/src/routes/error/error.jsx (hunk 17 above)
Comment (KonoganB) [reply genocrowd/react/src/routes/error/error.jsx]: aucune, j'ai fait la modification! mais je ne vois pas trop où j'utilise cette route pour l'instant de toute manière.

**Pull request merged**